packages/payload/src/utilities/configToJSONSchema.ts

```diff
@@ -70,50 +80,45 @@ function fieldsToJSONSchema(
   }
   required: string[]
 } {
-  // required fields for a schema (could be for a nested schema)
-  const requiredFields = new Set<string>(
-    fields.filter(propertyIsRequired).map((field) => (fieldAffectsData(field) ? field.name : '')),
-  )
+  const requiredFieldNames = new Set<string>()
```

Comment:
Any reason to use a `Set` instead of keyed object for the field names?

We don't often use these and the syntax differences/cognitive load hasn't seemed worth it to me, just curious.

Reply:
It was a Set before and I just did not want to refactor the below. The nice thing about them is they don't store duplicative values.

Reply:
Oh right, I didn't see that we also had the Set before.